packages/stk/stk_ngp_test/stk_ngp_test/GlobalReporter.cpp

```diff
@@ -24,11 +24,17 @@ ReporterBase*& getDeviceReporterOnHost()
 
 NGP_TEST_INLINE ReporterBase*& getDeviceReporterOnDevice()
 {
-  #ifdef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HIP_GPU
+#ifdef KOKKOS_ENABLE_HIP
+  KOKKOS_IF_ON_DEVICE((
     __device__ static ReporterBase* deviceReporterOnDevice = nullptr;
-  #else
+  ))
+  KOKKOS_IF_ON_HOST((
     static ReporterBase* deviceReporterOnDevice = nullptr;
-  #endif
+  ))
+#else
+  static ReporterBase* deviceReporterOnDevice = nullptr;
+#endif
+
```

Comment:
Wait did that actually compile?  `KOKKOS_IF_ON_{HOST,DEVICE}` is adding a scope, meaning I expect `deviceReporterOnDevice` is out of scope by the time you reach the return statement.

Reply:
No you are right. It wouldn't compile. I am unsure how it slipped through. I'll patch that in and update this PR.